src/main.rs

```diff
@@ -67,6 +67,28 @@ async fn main() -> Result<()> {
     Ok(())
 }
 
+fn setup_tracing() -> Result<()> {
+    let tracer = opentelemetry_otlp::new_pipeline()
+        .tracing()
+        .with_exporter(opentelemetry_otlp::new_exporter().tonic())
+        .install_batch(Tokio)?;
+    let tracing_layer = tracing_opentelemetry::layer().with_tracer(tracer).boxed();
+
+    let metrics = opentelemetry_otlp::new_pipeline()
+        .metrics(Tokio)
+        .with_exporter(opentelemetry_otlp::new_exporter().tonic())
+        .build()?;
+    let metrics_layer = MetricsLayer::new(metrics).boxed();
+
+    let filter = std::env::var("RUST_LOG").unwrap_or_else(|_| "info".to_string());
+    let env_filter = EnvFilter::builder().parse(filter)?;
+
+    let stacked_layer = tracing_layer.and_then(metrics_layer).and_then(env_filter);
+    tracing_subscriber::registry().with(stacked_layer).init();
+
+    Ok(())
+}
```

Comment:
Maybe we could add a bit of documentation on the different lines of this function to better understand the steps. As a beginner in the tracing domain these lines feel like quite obscure to me :)